internal/services/jsonrpc/usersvc/service.go

```diff
@@ -583,10 +583,12 @@ func (svc *Service) Call(ctx context.Context, req *userjson.CallRequest) (*userj
 		Dataset:   body.DBID,
 		Procedure: body.Action,
 		Args:      args,
-		TransactionData: common.TransactionData{
-			Signer:        signer,
-			Caller:        caller,
-			Height:        -1, // not available
+		TxCtx: &common.TxContext{
+			Signer: signer,
```

Comment:
Is it intended to keep `Ctx` nil here?  Is `common.TxContext.Ctx` only used in consensus / block execution code?  The `Ctx` field itself does not any documented conventions so the assumption is it should be set, I think

Reply:
Hmm that is an oversight. You are right that we are just using it to pass contextual vars. We could just duplicate the context to the TxContext? The other option is a larger breaking change to the engine interface by removing the first ctx parameter and replace it with tx context.

IMO the latter seems more logical.

Reply:
> The other option is a larger breaking change to the engine interface by removing the first ctx parameter and replace it with tx context.
> 
> IMO the latter seems more logical.

Yah, kinda.  Although looking at `svc.engine.Procedure`, this `TxContext` is just a field of `common.ExecutionData`, which is... not really a context?  Sorta.  It's partially arguments (dataset/procedure/args) and partially request-scoped stuff.  Does it feel a bit like `ExecutionData` went a little wrong in an effort to dedup?

Reply:
The way I see it:

An executions are within a transaction (there can be several executions in one tx in the case of a batch). Transactions are within a block, and blocks can be in a network (this is where the hierarchy comes from).

IMO, the most logical place for request scoping is within the `TxContext`. It seems like `ExecutionData` now should be `ExecutionContext`, and we replace the ctx.Context with these.

Reply:
I do agree that it seems like these shifted a bit though.

Reply:
Actually, we have a `ProcedureContext` which is essentially a duplicated version of `ExecutionData`. We only use it internally though

Reply:
I'm trying a few things out, will get something fixed for this tonight.

Reply:
> An executions are within a transaction (there can be several executions in one tx in the case of a batch). Transactions are within a block, and blocks can be in a network (this is where the hierarchy comes from).

Yeah, what you're doing along these lines makes perfect sense.  What feels less contexty is these:

```go
 		Dataset:   body.DBID,
 		Procedure: body.Action,
 		Args:      args,
```

I think it all works out fine. NBD.  As long as it's easy to manage passing of the actual `context.Context`.

Reply:
Yeah, what I am doing is keeping `ExecutionData` as just that, and passing the `TxContext` in place of the `context.Context`.